--- a/src/Omega_h_exodus.cpp
+++ b/src/Omega_h_exodus.cpp
@@ -12,6 +12,7 @@
 #include <algorithm>
 #include <iostream>
 #include <set>
+#include <sstream>
 
 #include "Omega_h_align.hpp"
 #include "Omega_h_array_ops.hpp"
@@ -74,9 +75,14 @@ static OMEGA_H_INLINE int side_exo2osh(
     case OMEGA_H_SIMPLEX:
       switch (dim) {
         case 2:
-          // seeing files from CUBIT with triangle sides in {3,4,5}...
-          // no clue what thats about, just modulo and move on
-          return (side) % 3;
+          switch (side) {
+            case 1:
+              return 0;
+            case 2:
+              return 1;
+            case 3:
+              return 2;
+          }
         case 3:
           switch (side) {
             case 1:
@@ -698,6 +704,11 @@ void write(
     }
     std::vector<char*> set_name_ptrs(surface_set.size(), nullptr);
     for (std::size_t i = 0; i < set_names.size(); ++i) {
+      if (set_names[i].empty()) {
+        std::stringstream ss;
+        ss << "surface_" << i;
+        set_names[i] = ss.str();
+      }
       set_name_ptrs[i] = const_cast<char*>(set_names[i].c_str());
     }
     if (classify_with & exodus::NODE_SETS) {
