--- a/src/Omega_h_exodus.cpp
+++ b/src/Omega_h_exodus.cpp
@@ -74,9 +74,14 @@
     case OMEGA_H_SIMPLEX:
       switch (dim) {
         case 2:
-          // seeing files from CUBIT with triangle sides in {3,4,5}...
-          // no clue what thats about, just modulo and move on
-          return (side) % 3;
+          switch (side) {
+            case 1:
+              return 0;
+            case 2:
+              return 1;
+            case 3:
+              return 2;
+          }
         case 3:
           switch (side) {
             case 1:
